core/factorization/ic.cpp

```diff
@@ -104,7 +104,7 @@ std::unique_ptr<Composition<ValueType>> Ic<ValueType, IndexType>::generate(
     std::shared_ptr<const matrix_type> ic;
     // Compute LC factorization
     if (parameters_.algorithm == incomplete_algorithm::syncfree ||
-        exec == exec->get_master()) {
+        !std::dynamic_pointer_cast<const ReferenceExecutor>(exec)) {
```

Comment:
I think you still need the check for `OmpExecutor`, because otherwise this will also trigger for cuda/hip/dpcpp executor with sparselib selected.